src/components/__tests__/DisciplineGroup.tsx

```diff
@@ -0,0 +1,92 @@
+  climbTypes = value
+}
+
+describe('DisciplineGroup', () => {
+  beforeEach(() => {
+    climbTypes = {
+      trad: true,
+      sport: true,
+      tr: true,
+      bouldering: true
+    }
+  })
+  it('renders component without error', () => {
+    render(<DisciplineGroup defaultTypes={defaultTypes} climbTypes={climbTypes} setClimbTypes={setClimbTypes} />)
+  })
+
+  it('all 4 types selected by default', () => {
+    const { getByRole } = render(<DisciplineGroup defaultTypes={defaultTypes} climbTypes={climbTypes} setClimbTypes={setClimbTypes} />)
+    const sportButton = getByRole('checkbox', { name: /Sport/i })
+    const tradButton = getByRole('checkbox', { name: /Trad/i })
+    const topRopeButton = getByRole('checkbox', { name: /Top rope/i })
+
+    expect(sportButton).toBeChecked()
+    expect(tradButton).toBeChecked()
+    expect(topRopeButton).toBeChecked()
+  })
+
+  it('changes values and resets to defaultTypes when component regenerated', async () => {
+    const clickButton = async (button): Promise<void> => {
+      await userEvent.click(button)
+      rerender(<DisciplineGroup defaultTypes={defaultTypes} climbTypes={climbTypes} setClimbTypes={setClimbTypes} />)
+    }
+
+    const { container, getByRole, rerender, unmount } = render(<DisciplineGroup defaultTypes={defaultTypes} climbTypes={climbTypes} setClimbTypes={setClimbTypes} />)
+    expect(container.getElementsByClassName('border-neutral-800').length).toBe(4)
+
+    const sportButton = getByRole('checkbox', { name: /Sport/i })
+    const tradButton = getByRole('checkbox', { name: /Trad/i })
+    const topRopeButton = getByRole('checkbox', { name: /Top rope/i })
+
+    // click buttons to remove from filter
+    // sportButton
+    expect(sportButton).toBeChecked()
+    await clickButton(sportButton)
+    expect(sportButton).not.toBeChecked()
+
+    // tradButton
+    expect(tradButton).toBeChecked()
+    await clickButton(tradButton)
+    expect(tradButton).not.toBeChecked()
+
+    // topRopeButton
+    expect(topRopeButton).toBeChecked()
+    await clickButton(topRopeButton)
+    expect(topRopeButton).not.toBeChecked()
+    await clickButton(topRopeButton)
+    expect(topRopeButton).toBeChecked()
+
+    // cancel and click away functionality.
+    unmount() // cancel/click-away unmounts <DisciplineGroup /> component
+
+    const { rerender: rerender2, getByRole: getByRole2 } = render(<DisciplineGroup defaultTypes={defaultTypes} climbTypes={climbTypes} setClimbTypes={setClimbTypes} />)
+
+    // rerender needed so that useEffect [] runs.
+    rerender2(<DisciplineGroup defaultTypes={defaultTypes} climbTypes={climbTypes} setClimbTypes={setClimbTypes} />)
+
+    const sportButton2 = getByRole2('checkbox', { name: /Sport/i })
+    const tradButton2 = getByRole2('checkbox', { name: /Trad/i })
+    const topRopeButton2 = getByRole2('checkbox', { name: /Top rope/i })
+    const boulderingButton2 = getByRole2('checkbox', { name: /Bouldering/i })
```

Comment:
you should not need to redeclare these right?
we re-rendered but it is still the same element I would imagine it updated?

Reply:
it seems that they still have the old values even after explicitly unmounting the component, which is why i redeclared the variables. from what i can tell these are just stale values. do you think that could indicate an issue?

Reply:
as long as you checked, fine by me

Reply:
you can change the const to let then and reassign those buttons

that would make it more copy and paste friendly.
however, having seperate consts is safer.

there are merrits to both ways.

these are just suggestions to get you to think about it. feel free to disregard.